ide/lsp.client/src/org/netbeans/modules/lsp/client/debugger/Utils.java

```diff
@@ -0,0 +1,127 @@
+import org.openide.text.Line.ShowVisibilityType;
+
+public final class Utils {
+
+    private static final Logger logger = Logger.getLogger(Utils.class.getName());
+
+    private static Object currentLine;
+
+    private Utils() {
+    }
+
+    static synchronized void markCurrent (final Object line) {
+        unmarkCurrent ();
+
+        Annotatable[] annotatables = (Annotatable[]) line;
+        int i = 0, k = annotatables.length;
+
+        // first line with icon in gutter
+        DebuggerAnnotation[] annotations = new DebuggerAnnotation [k];
+        if (annotatables [i] instanceof Line.Part)
+            annotations [i] = new DebuggerAnnotation (
+                DebuggerAnnotation.CURRENT_LINE_PART_ANNOTATION_TYPE,
+                annotatables [i]
+            );
+        else
+            annotations [i] = new DebuggerAnnotation (
+                DebuggerAnnotation.CURRENT_LINE_ANNOTATION_TYPE,
+                annotatables [i]
+            );
+
+        // other lines
+        for (i = 1; i < k; i++)
+            if (annotatables [i] instanceof Line.Part)
+                annotations [i] = new DebuggerAnnotation (
+                    DebuggerAnnotation.CALL_STACK_FRAME_ANNOTATION_TYPE,
+                    annotatables [i]
+                );
+            else
+                annotations [i] = new DebuggerAnnotation (
+                    DebuggerAnnotation.CALL_STACK_FRAME_ANNOTATION_TYPE,
+                    annotatables [i]
+                );
+        currentLine = annotations;
+
+        showLine (line);
+    }
+
+    static synchronized void unmarkCurrent () {
+        if (currentLine != null) {
+
+//            ((DebuggerAnnotation) currentLine).detach ();
+            int i, k = ((DebuggerAnnotation[]) currentLine).length;
+            for (i = 0; i < k; i++) {
+                ((DebuggerAnnotation[]) currentLine) [i].detach ();
+            }
+            currentLine = null;
+        }
+    }
+
+    public static void showLine (final Object line) {
+        final Annotatable[] a = (Annotatable[]) line;
+        SwingUtilities.invokeLater (new Runnable () {
+            @Override
+            public void run () {
+                if (a[0] instanceof Line) {
+                    ((Line) a[0]).show (ShowOpenType.OPEN, ShowVisibilityType.FOCUS);
+                } else if (a[0] instanceof Line.Part) {
+                    ((Line.Part) a[0]).getLine ().show (ShowOpenType.OPEN, ShowVisibilityType.FOCUS);
+                } else {
+                    throw new InternalError(a[0].toString());
+                }
+            }
+        });
+    }
+
+    static int getLineNumber (Object line) {
+        final Annotatable[] a = (Annotatable[]) line;
+        if (a [0] instanceof Line)
+            return ((Line) a [0]).getLineNumber ();
+        else
+        if (a [0] instanceof Line.Part)
+            return ((Line.Part) a [0]).getLine ().getLineNumber ();
+        else
+            throw new InternalError ();
+    }
```

Comment:
Unused

Reply:
Removed in ad198730ef6fc90fc04ed4cad855e629be3006cb